src/main/java/toothless/Parser.java

```diff
@@ -0,0 +1,92 @@
+package toothless;
+
+import toothless.commands.ByeCommand;
+import toothless.commands.Command;
+import toothless.commands.DeadlineCommand;
+import toothless.commands.DeleteCommand;
+import toothless.commands.EventCommand;
+import toothless.commands.FindCommand;
+import toothless.commands.ListCommand;
+import toothless.commands.MarkCommand;
+import toothless.commands.TodoCommand;
+import toothless.commands.UnmarkCommand;
+
+/**
+ * This class is responsible for interpreting the user's inputs and converting them
+ * into specific Commands that can be executed within the application.
+ * This class acts as a bridge between the user interface and the application's command execution logic,
+ * ensuring that user inputs are understood and acted upon appropriately.
+ */
+public class Parser {
+
+    /**
+     * Parses the given input string to produce a corresponding Command object that represents
+     * a specific operation or action to be performed by the application.
+     * This method identifies the type of command along with any necessary arguments by the user.
+     * @param input the raw input string provided by the user, which includes the command keyword and any arguments
+     * @return Command corresponding to the user's request
+     * @throws ToothlessException if the input is invalid or no Command is suitable
+     */
+    public static Command parseCommand(String input) throws ToothlessException {
```

Comment:
Good  idea of grouping different commands into groups for use of same parser commands. Will take inspo!